internal/ast/ast.go

```diff
@@ -2353,6 +2353,9 @@ func IsArrayLiteralOrObjectLiteralDestructuringPattern(node *Node) bool {
 
 func accessKind(node *Node) AccessKind {
 	parent := node.Parent
+	if parent == nil {
+		return AccessKindRead
+	}
```

Comment:
Ah yes, Strada said:

```ts
function accessKind(node: Node): AccessKind {
    const { parent } = node;

    switch (parent?.kind) {
```

Love the ol "parent is never undefined except when it totally is"